### PR TITLE
Create AprilTagLocalizer

### PR DESCRIPTION
**Pull request by drwho495**
This localizer is mostly complete, I will take it out of draft when I have fully verified its functionality

Thanks to [@j5155](https://github.com/j5155) for some of the code that finds the position of the robot, and others.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+ *    \--------------/
+ *      front (x pos)
+ *
+ * @author Anyi Lin - 10158 Scott's Bots
+ * @version 1.0, 7/20/2024
+ */
+
+/**
+ * Copyright (c) 2024, Austin Bacher @hypocritical1
+ * All rights reserved.
```
Comment (j5155) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer:52]: Change this to the name of the license you are releasing under. Looks like you're using BSD-3-Clause?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+        // rotate RC coordinates to be field-centric
+        double x2 = x*Math.cos(botheading)+y*Math.sin(botheading);
+        double y2 = x*-Math.sin(botheading)+y*Math.cos(botheading);
+        // add FC coordinates to apriltag position
+        // tags is just the CS apriltag library
+        VectorF tagpose = getCenterStageTagLibrary().lookupTag(detection.id).fieldPosition;
+
+
+        // todo: this will need to be changed for next season (use tag heading to automate??)
```
Comment (j5155) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer:183]: I never changed this for this season, so this assumes you're still in Centerstage tags. You'll need to update it.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+    }
+
+    @Override
+    public double getTurningMultiplier() {
+        return backupLocalizer.getTurningMultiplier();
+    }
+
+    // this custom position library credit Michael from team 14343 (@overkil on Discord)
+    // TODO: will need to be changed for 24-25 season
+    public AprilTagLibrary getCenterStageTagLibrary()
```
Comment (j5155) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer:282]: This is no longer accurate for this season.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer (hunk 2 above)
Comment (drwho495) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/pedroPathing/localization/localizers/AprilTagLocalizer:183]: Yeah I'm aware, I haven't had a chance to finish this, but I will try to get it done eventually.

**Comment by Logan-Nash** [comment]
as previously discussed, for now I do not see a future of adding an apriltag localizer currently. As we transition to a library that is something we may add in the future. Not enough users are interested due to apriltag positioning this year.

**Pull request closed without merging**